### ISSUE TITLE
edit-and-restart option

### ISSUE THREAD

**Issue by sourcefrog** [Nukesor/pueue#83]
If I make a mistake in a command I currently need to, I think, `pueue restart --stashed` then `pueue edit` then `pueue enqueue`. (Or retype or copy paste the command.)

It'd be nice if there was a `pueue restart --edit` that gave me a chance to edit before restarting it.

**Comment by Nukesor** [comment]
I see how this can be helpful and I recall a few occasions where I could have used something like this.

The only reason I'm hesitant is, that this requires some ping-pong between the daemon and the client, which cannot be easily handled right now. The protocol is really simple...

Best way would probably be to -> 
1. Request the task from the daemon
2. Create a new task from it
3. Edit path/command
4. Add new task to daemon

An alternative would be to:
1. Send restart request to daemon, but force the task to instantly enter Locked status on creation.
2. Continue using the open socket to request the path/command
3. Follow the normal edit steps

I'm not sure which of these is the best. Both seem a little hacky

**Comment by sourcefrog** [comment]
Not sure if this is the same as your first option, but what I'd be inclined to do is:

1. Client reads the existing task command from the server
2. Pop that string into an editor and let the user change it. 
3. Create a new task with the given command.

**Comment by Nukesor** [comment]
Jep. That's the same as my first option :D

**Comment by Nukesor** [comment]
This feature is now implemented on this [branch](https://github.com/Nukesor/pueue/tree/client_refactoring), but it is completely untested. (I just now managed to get it compiling. Would love to test it, but I gotta sleep)

While implementing the feature, the Client got a pretty big structural refactoring. The old code was a little chaotic, since it has never been designed for this level of complexity.
